src/components/SocialMediaLinks/SocialMediaLinks.tsx

```diff
@@ -15,24 +14,33 @@ export type ColorType = typeof colorTypeArray[number];
 export const sizeTypeArray = ["small", "medium", "large"] as const;
 export type SizeType = typeof sizeTypeArray[number];
 
-// This list of allowed types is defined in the SocialMediaDataMap found in SocialMediaLinksUtils.ts and extracted here.
-export const socialMediaLinkTypeArray = socialMediaDataMap.map(
-  ({ type }) => type
-);
-export type SocialMediaLinkType = typeof socialMediaLinkTypeArray[number];
+type IconNames =
+  | "fileTypeGenericDoc"
+  | "socialFacebook"
+  | "socialInstagram"
+  | "socialPinterest"
+  | "socialSoundCloud"
+  | "socialTikTok"
+  | "socialTumblr"
+  | "socialTwitter"
+  | "socialYoutube";
+
+// Verify that SocialMediaIconNames is a subset of IconNames
+type VerifyIconNames = IconNames extends DsIconNames ? true : false;
+// TYPE SAFETY: verified will throw an error if a value has been added to socialMediaIconMap that is not a IconName
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const verified: VerifyIconNames = true;
 
 export interface SocialMediaLinkDataProps {
-  /** Optional override for default platform name */
-  labelText?: string;
+  /** Required. Label rendered on page is showLabel is true  */
+  labelText: string;
   /** Required. Must be one of socialMediaLinkTypeArray */
-  type: SocialMediaLinkType;
-  /** Optional override for default social media url */
-  url?: string;
+  type: IconNames;
+  /** Required. Url to link to the platform */
+  url: string;
 }
 
-export interface SocialMediaLinksProps {
-  /** Optional border: straight, circular or none. */
-  borders?: BorderType;
+interface CommonSocialMediaLinksProps {
```

Comment:
nit: our convention is to call this "base" instead of "common"

Reply:
Changed.